### DISCUSSION TITLE
Template loading order according to composer.json dependencies

### DISCUSSION THREAD

**Discussion by Schrank** [shopware/platform#3154] [RFC / Proposal]
**Effort**: high

### Description: 
Shopware should solve the dependencies between plugins (defined by composer.json) before deciding on the order of templates-inheritance processed

### Benefits
Currently it is broken.
You install Plugin A today and Plugin B which overwrites something in B tomorrow. The template from B is not picked up, if it is inherited from A. If you change plugin.created_at in the database, so B's created_at is "older" (last week) than A everything works as expected. But the order shouldn't be dependant of any install date but on the dependency defined by composer.

### Break strategy
Imho there is no break strategy needed, because currently the behaviour is broken and useless unless you do a "core hack" by changing the database.

**Comment by shyim** [comment]
Funny built similar thing 4 years ago for 5 https://gist.github.com/shyim/ecefe9b701f3dc2c79a716f4f449545c 😅

**Comment by Schrank** [comment]
That looks like Effort: low 🎉

**Comment by mitelg** [comment]
maybe there is something with composer already? could be done with shopware/proposal#14 maybe? 🤔

**Comment by AndreasA** [comment]
Not sure if this is already done but it probably should also be how the plugin config, plugin extensions etc. are loaded, e.g. relevant for https://symfony.com/doc/current/bundles/prepend_extension.html#more-than-one-bundle-using-prependextensioninterface

But maybe provide an extra property to change a plugins priority too ? though probably dependency is enough.
